pkg/providers/v2/interface.go

```diff
@@ -41,6 +48,86 @@ var _ cloudprovider.Interface = (*cloud)(nil)
 type cloud struct {
 }
 
+type awsSDKProvider struct {
+	creds *credentials.Credentials
+}
+
+// EC2Metadata is an abstraction over the AWS metadata service.
+type EC2Metadata interface {
+	// Query the EC2 metadata service (used to discover instance-id etc)
+	GetMetadata(path string) (string, error)
+}
+
+func getAvailabilityZone(metadata EC2Metadata) (string, error) {
+	return metadata.GetMetadata("placement/availability-zone")
+}
+
+// Derives the region from a valid az name.
+// Returns an error if the az is known invalid (empty)
+func azToRegion(az string) (string, error) {
```

Comment:
Can we add a unit test for this?

Reply:
sure, will add soon!

Reply:
@andrewsykim Updated!
I add a new file called interface_test.go and include the unit test for it
I also delete the type awsSDKProvider and call metadataClient directly in newCloud()